packages/connect-express/package.json

```diff
@@ -20,6 +20,10 @@
   "main": "./dist/cjs/index.js",
   "exports": {
     ".": {
+      "node": {
+        "require": "./dist/cjs/index.js",
+        "import": "./dist/proxy/index.js"
+      },
       "module": "./dist/esm/index.js",
       "require": "./dist/cjs/index.js",
       "import": "./dist/proxy/index.js"
```

Comment:
This doesn't look right. The point of adding the `node` condition is only use the ESM wrapper on Node.js, because it will fail for other ESM consumers. The ESM wrapper simply delegates to CJS, but only Node.js can import CJS from ESM.

You still have the `import` condition pointing to the ESM wrapper. ESM consumers that don't enter the `node` condition will use this, and fail:

```
"import": "./dist/proxy/index.js"
```

See https://github.com/bufbuild/protobuf-es/issues/610#issuecomment-1802152966

Reply:
I see. So, the plain `import` condition **not** nested inside the `node` condition should point to `./dist/esm/index.js`?

Note that #610 is also doing the same thing I'm doing here. I'm assuming that should also be changed?

Reply:
> I see. So, the plain import condition **not** nested inside the node condition should point to ./dist/esm/index.js?

Correct. 

> Note that https://github.com/bufbuild/protobuf-es/issues/610#issuecomment-1802152966 is also doing the same thing I'm doing here. I'm assuming that should also be changed?

Do you mean the exports in my comment, or https://github.com/bufbuild/protobuf-es/pull/611?

If possible, it's best to keep changes to a minimum. So make the change here, test it in examples-es, _then_ come back to https://github.com/bufbuild/protobuf-es/pull/611.

Reply:
I mean the actual changes as implemented in bufbuild/protobuf-es#611. They also specify a plain import condition pointing to the ESM wrapper.

Reply:
Updated the paths for the top-level import condition. If this looks good, I'll create the PR for examples-es.

Reply:
Here's what this uses now:

```
      "node": {
        "require": "./dist/cjs/index.js",
        "import": "./dist/proxy/index.js"
      },
      "module": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js",
      "import": "./dist/esm/index.js"
```

Which is not wrong, but also not ideal, for the following reasons:

1. We should be able  to remove `module` completely here. Even from the [webpack docs](https://webpack.js.org/guides/package-exports/#reference-syntax), where this condition originates from, I don't see a reason we should need it as long as we have the `import` condition.
2. Since order matters, and we certainly prefer `import` over `require`, `import` should come first, before `require`.
3. `node.require` seems irrelevant. I would expect a CJS request from Node.js to match the `require` condition. So we should be able to remove this.

Applying these remarks, we end up with:

```
      "node": {
        "import": "./dist/proxy/index.js"
      },
      "import": "./dist/esm/index.js", 
      "require": "./dist/cjs/index.js",
```

This differs from the suggestion in https://github.com/bufbuild/protobuf-es/issues/610#issuecomment-1802152966 to respect ordering, see point 2.